### PR TITLE
DUPLO-25488: dynamo tbl tag based issue

### PR DESCRIPTION
**Pull request by pravin-duplocloud**
## Overview

DUPLO-25488: dynamo tbl tag based issue (Testing)

## Summary of changes

This PR does the following:

- use fullname as table name in duplo crud api

## Testing performed

- [ ] Using unit tests
- [X] Manually, on my local system
- [X] Manually, on a remote test system

## Describe any breaking changes

- None

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-25488 TF: Dynamo Naming causing tables to get created instead of updated](https://app.clickup.com/t/8655600/DUPLO-25488)

**Comment by pravin-duplocloud** [comment]
TF
1. https://github.com/duplocloud/terraform-provider-duplocloud/pull/729 
2. bug/DUPLO-25488-dynamo-tbl-tag-based-issue

BE
1.  https://github.com/duplocloud-internal/duplo/pull/3280 
2. bug/DUPLO-25518-dynamo-tbl-tag-based-issue (based on master)

tested 1
1. TagBasedResources disabled. <add key="ENABLEAWSRESOURCEMGMTUSINGTAGS" value="false" />
2. create from UI 
3. create from TF (tf apply again) 


tested 2
1. TagBasedResources enabled. <add key="ENABLEAWSRESOURCEMGMTUSINGTAGS" value="true" />
2. create from UI 
3. create from TF (tf apply again)

File: duplocloud/resource_duplo_aws_dynamodb_table_v2.go [hunk 1]
```diff
@@ -488,18 +480,19 @@ func resourceAwsDynamoDBTableCreateV2(ctx context.Context, d *schema.ResourceDat
...
+
 func updateDynamoDBTTl(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	if v, ok := d.GetOk("ttl"); ok {
 		id := d.Id()
 		tenantID, name, err := parseAwsDynamoDBTableIdParts(id)
 		if err != nil {
 			return diag.Errorf("Error while parsing id %s", err.Error())
 		}
-		fname := d.Get("fullname").(string)
+		fullname := d.Get("fullname").(string)
```
Comment (Duplo-nikv) [inline duplocloud/resource_duplo_aws_dynamodb_table_v2.go:492]: do we need this now?

File: duplocloud/resource_duplo_aws_dynamodb_table_v2.go [hunk 2]
```diff
@@ -511,160 +504,16 @@ func updateDynamoDBTableV2PointInRecovery(_ context.Context, d *schema.ResourceD
...
-		r.TableName = fullname
-
-		_, err := c.DuploDynamoDBTableV2UpdateDeletionProtection(tenantID, &r)
-		if err != nil {
-			return diag.FromErr(err)
-		}
-	}
-	return nil
-}*/
-
```
Comment (Duplo-nikv) [inline duplocloud/resource_duplo_aws_dynamodb_table_v2.go:667]: We can use this commented update context if backend can support this features

Comment (pravin-duplocloud) [reply duplocloud/resource_duplo_aws_dynamodb_table_v2.go:667]: sure ..   lets you move that into next pr ... will do any BE changes if required

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**